Refactor PV Excess Optimizer blueprint to add sections

HAuser1234 opened the pull request:
Rearranged and added sections to the PV Excess Optimizer blueprint YAML file to improve clarity and organization. I found it quite hard to find what you need quickly so I decided to add some sections to it :)
I hope you all like it.

HAuser1234 committed:
Refactor PV Excess Optimizer blueprint to add sections

Updated the description and input fields in the PV Excess Optimizer blueprint YAML file to improve clarity and organization.

tabascoz commented:
thanks for the contribution! will give a look the weekend and revert

HAuser1234 commented:
Great! let me know if you find any mistakes I made. I have this version now in my system and it seems to work :)

HAuser1234 committed:
Update pv_excess_control.yaml with execution note to prevent average power calculation errors

Added note about minimum time requirement for automation execution.

Kolbi commented:
Thanks for doing this, I never had the time to do it :D As we add more and more functions and parameters it's getting more and more messy.

@tabascoz did you had the chance to check the pr?

tabascoz commented:
I'm checking it now

tabascoz commented:
Some proposed changes: 

1. blueprint's name, remove INVENTOCASA, let it be just "PV excess optimizer" I think the proper credit has to be done, but the script  has been diverging enough to justify a rebrand. 
2. For "export power" must worth mentioning that for zero-feedin this sensor should be popoulated with a dummy "zero-value" sensor as you'll never be exporting. Perhaps it needs a note in the documentation. I've created a dummy template sensor like  below: 
```
Sensor: 
  - name: "grid_energy_out"
    unit_of_measurement: "kWh"
    state_class: total_increasing
    device_class: energy
    state: 0
```
Another approach would be caring for this variable at the python code ( it might be better and just ignore the value - easy for the user. @Kolbi  your thoughts here i can do PR and submit. 

3. Although the organization it feels right the way it is, i was wondering if "appliance priority" should be at the top in general settings section, given it's importance. either way it works, but I'd suggest to change. 

4. Enable Device Switch remove "optional" ( i don't think this is indeed optional ) and move it to the top of the section. 

Cheers, Tabascoz

HAuser1234 committed:
Rename blueprint to 'PV Excess Optimizer'

HAuser1234 committed:
Update pv_excess_control.yaml with zero-feedin note

Added instruction for zero-feedin configuration.

HAuser1234 committed:
Add appliance priority setting to the general section

Added appliance priority configuration for better control over appliance switching based on priority levels.

HAuser1234 committed:
move device enable/disable switch to top of appliance settings

Added an optional switch to enable or disable the device.

HAuser1234 commented:
@tabascoz Thank you for the review! I changed some things based on your feedback. Are you sure it the Enable Device Switch is not optional it seems to work without it defined :)

tabascoz commented:
> @tabascoz Thank you for the review! I changed some things based on your feedback. Are you sure it the Enable Device Switch is not optional it seems to work without it defined :)

Sorry! you're right. Aside of being a little tired yesterday, i think also the description it wasn't clear enough. I read it yesterday and thought it was to define the device switch, but it is indeed the automation kill switch which competes disable device actions. 

Therefore i agree to move it to the top and rename it / change the description to something like : 

- Automation kill switch 

- Use this switch when you need the automation to stop managing your device's power, allowing you to temporarily pause automation and regain manual control. When disabled, the automation stops making changes, leaving the device in its current state.

HAuser1234 commented:
@tabascoz how about this desciption:

          name: Enable Device Switch
          description: 'Optional entity to enable/disable this appliance automation.

           allows you to enable / disable the device control without an extra automation needed - just add a binary state sensor!

            **When ON/True**: Appliance will be controlled automatically based on PV excess power
            **When OFF/False**: Appliance will be completely ignored by the automation

            **Compatible entities:**
            - **input_boolean**: Manual dashboard control
            - **switch**: Physical or virtual switches  
            - **binary_sensor**: Presence sensors, occupancy detection, etc.

            **Examples:**
            - `input_boolean.pool_pump_enabled` - Manual control
            - `binary_sensor.family_home` - Only run when someone is home
            - `binary_sensor.workday` - Only run on workdays


Isn't the real kill switch supposed to be the automation beeing disabled or enabeld?

tabascoz commented:
> @tabascoz how about this desciption:
> 
> ```
>       name: Enable Device Switch
>       description: 'Optional entity to enable/disable this appliance automation.
> 
>        allows you to enable / disable the device control without an extra automation needed - just add a binary state sensor!
> 
>         **When ON/True**: Appliance will be controlled automatically based on PV excess power
>         **When OFF/False**: Appliance will be completely ignored by the automation
> 
>         **Compatible entities:**
>         - **input_boolean**: Manual dashboard control
>         - **switch**: Physical or virtual switches  
>         - **binary_sensor**: Presence sensors, occupancy detection, etc.
> 
>         **Examples:**
>         - `input_boolean.pool_pump_enabled` - Manual control
>         - `binary_sensor.family_home` - Only run when someone is home
>         - `binary_sensor.workday` - Only run on workdays
> ```
> 
> Isn't the real kill switch supposed to be the automation beeing disabled or enabeld?

Pehaps "killswitch" might be too strong ( this is the way that i use it ) but it doesn't actually disable the automation. Per ex even with it disabled the automation still process and accounts data, it does only stay inert and does not do any actions. 

So it's somewhere in the middle ( lol) think your proposal is right on spot.

HAuser1234 committed:
Enhance description for Enable Device Switch option

Updated the description for the Enable Device Switch option to provide detailed information on its usage, compatible entities, and examples.

HAuser1234 commented:
@tabascoz @Kolbi I've now implemented all the requested enhacements. I use the current version in my setup and it seems to work! Are there any more changes needed?

tabascoz commented:
> @tabascoz @Kolbi I've now implemented all the requested enhacements. I use the current version in my setup and it seems to work! Are there any more changes needed?

I tested here and it's all good. Now goes to @Kolbi  for approval :) Thanks

Kolbi commented:
Looks good so far, I just want to add some pre-commit "optical" changes later on, before merging.

Kolbi committed:
Update pv_excess_control.yaml

Kolbi commented:
Only thing to complain yamllint again has a lot of lines with 

> error    line too long (86 > 80 characters)  (line-length)
https://results.pre-commit.ci/run/github/942486024/1759906650.Do-YedklT5-zM_rBKuiAUg

Kolbi commented:
@HAuser1234 have you tried using "enabled" with a binary_sensor?

`**Examples:**
            - `input_boolean.pool_pump_enabled` - Manual control
            - `binary_sensor.family_home` - Only run when someone is home
            - `binary_sensor.workday` - Only run on workdays

            "
          default:
          selector:
            entity:
              domain:
                - input_boolean
                - switch`

Currently only switch and input_boolean are tested by myself but I would add binary_sensor as well if tested and working.

HAuser1234 commented:
@Kolbi I have one binary_sensor currently in one of the automations and it seems to work :)
